canonical-web-and-design/cloud-init.io
# Switch Jekyll to eleventy

Remove need for Ruby dependencies, and add `dotrun` support.

This change hasn't been discussed widely, but `dotrun` can't support Ruby, and anyway our team has no expertise in Ruby. Eleventy is a pretty close match to Jekyll - most things work almost out of the box - so it shouldn't be too jarring.

QA
--

**Note:** For the time being the circleci tests will fail, as I removed them and replaced them with GitHub actions. When this PR is approved, I'll update the branch protection rules so this can be merged.

```
dotrun
```

Go to http://localhost:8009, see the site working.

```
docker build -t ci .
docker run -ti -p 8888:80 ci
```

Go to http://localhost:8888, see the site working. Try http://localhost:8888/non-existent-page, see the 404 page.
